client/data/disputes/hooks.ts

```diff
@@ -5,8 +5,14 @@
  */
 import { useSelect, useDispatch } from '@wordpress/data';
 import { STORE_NAME } from '../constants';
+import type { Dispute } from 'wcpay/types/disputes';
 
-export const useDispute = ( id ) => {
+export type UseDisputeObject = {
```

Comment:
@KarlisJ, I put more thought here and suggest following this approach:

```javascript
export const useDispute = (
	id: string
): {
	dispute: Dispute;
	isLoading: boolean;
	doAccept: () => void;
} => {
```
That allows to keep high-level types, but reduces the number of local ones. Also less coupling between components and the data layer.

@anu-rock , @dmvrtx what's your thought here?

Reply:
other changes looks good so far.

Reply:
I'm more in favour of defining local interfaces. My reasoning is to reduce duplication of definitions between producers and consumers.

Reply:
Sounds good to me. It's one of those situations where an "inlined" type is better than a named type because of the amorphous nature of hooks.

Reply:
I did this (code example from Vlad) in my first iteration and disliked the code duplication. For this case that might be acceptable as a general rule it could create some serious code depth to keep track of types.

Reply:
I understand, but this object is too specific for the data layer. Other layers should definitely take advantage of `Dispute`, but this object shouldn't;t imported in `client/disputes/info/index.tsx`.

When use use the result in the application code, you pick the only part you need - it's similar to PHP's array restructuring.

Reply:
Updated in [#4ac6cf8b79518bdb1d77925c0605604584e6956a](https://github.com/Automattic/woocommerce-payments/pull/3491/commits/4ac6cf8b79518bdb1d77925c0605604584e6956a)

Reply:
Thanks!